vizro-core/tests/unit/vizro/models/_components/form/test_radioitems.py

```diff
@@ -0,0 +1,144 @@
+            (
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}],
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}],
+            ),
+            (
+                [{"label": "New York", "value": "NYC"}, {"label": "Berlin", "value": "BER"}],
+                [{"label": "New York", "value": "NYC"}, {"label": "Berlin", "value": "BER"}],
+            ),
+            (
+                [{"label": "True", "value": True}, {"label": "False", "value": False}],
+                [{"label": "True", "value": True}, {"label": "False", "value": False}],
+            ),
+            ([True, 2.0, 1.0, "A", "B"], ["True", "2.0", "1.0", "A", "B"]),
+        ],
+    )
+    def test_create_radio_items_valid_options(self, test_options, expected):
+        radio_items = RadioItems(options=test_options)
+
+        assert hasattr(radio_items, "id")
+        assert radio_items.type == "radio_items"
+        assert radio_items.options == expected
+        assert radio_items.value is None
+        assert radio_items.title is None
+        assert radio_items.actions == []
+
+    @pytest.mark.parametrize("options", [1, "A", True, 1.0])
+    def test_create_radio_items_invalid_options_type(self, options):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
+            RadioItems(options=options)
+
+    def test_create_radio_items_invalid_options_dict(self):
+        with pytest.raises(
+            ValidationError,
+            match="Invalid argument `options` passed into Selector. Expected a dict with keys `label` and `value`.",
+        ):
+            RadioItems(options=[{"hello": "A", "world": "A"}, {"hello": "B", "world": "B"}])
+
+    @pytest.mark.parametrize(
+        "options, test_value",
+        [
+            (["A", "B", "C"], "A"),
+            ([1, 2, 3], 1),
+            ([1.0, 2.0, 3.0], 2.0),
+            ([True, False], True),
+            ([{"label": "A", "value": "A"}, {"label": "B", "value": "B"}], "B"),
+            ([True, 2.0, 1.0, "A", "B"], "True"),
+        ],
+    )
+    def test_create_radio_items_valid_value(self, options, test_value):
+        radio_items = RadioItems(options=options, value=test_value)
+
+        assert hasattr(radio_items, "id")
+        assert radio_items.type == "radio_items"
+        assert radio_items.value == test_value
+        assert radio_items.title is None
+        assert radio_items.actions == []
+
+    @pytest.mark.parametrize(
+        "test_value, options",
+        [
+            (5, ["A", "B", "C"]),
+            (5, [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}]),
+            ("D", ["A", "B", "C"]),
+            ("D", [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}]),
+            (True, ["A", "B", "C"]),
+            (True, [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}]),
+        ],
+    )
+    def test_create_radio_items_invalid_value_non_existing(self, test_value, options):
```

Comment:
Can we preserve parameter order in those two tests (or 
 - `"options, test_value",` or 
 - `"test_value, options",`
 ?